hicommonwealth/commonwealth
# Cache and lazy-load Cosmos queries

## Link to Issue
Closes: #4527 

## Description of Changes
- Moves more controller fetches to lazy-load with React Query:
  - specific proposal, votes, tally, deposits, metadata
  - Chain Params for deposit, staking, pool 
 - Updates UI to react to new loading flow. Expect to see spinner less, and useful Views faster, especially after first load. 

## "How We Fixed It"


## Test Plan
- ran proposals.spec.ts E2E tests locally
- CA (click around) tested on local and frack:
  - clear cache
  - Go to /osmosis and /juno and join communities (osmosis covers v1beta1, juno covers v1)
  - on Osmosis main page, click "Proposals" 
    - confirm active and inactive proposals load
    - confirm status loads async (Passing, Failing, etc should load after proposal cards are visible)
  - Click an individual proposal
    - confirm Proposal page loads correctly
    - confirm that votes load async
    - (V1 only): confirm that Proposal titles that rely on metadata load async (Juno Prop 316 for example)
  - Click back to "Proposals"
    - confirm that Proposal Cards are instantly visible
    - confirm that Status text updates if needed
  - Using Explore sidebar, navigate to Juno and repeat above tests for this community
  - Using Explore sidebar, go back to Osmosis -> Proposals
    - Confirm that proposal Cards visible almost instantly (they are cached, and updates are fetched in the background)
  - Using the [shared devnet account](https://github.com/hicommonwealth/commonwealth/blob/master/wiki/Devnet.md#how-to-manually-test-transactions-on-the-csdk-or-csdk-beta-sandbox-community), login to /csdk
    - Create a proposal, then immediately click to "Proposals"
      - confirm the active proposal is visible within a few seconds
      - Click back to the active proposal. Wait for the proposal voting period (90 seconds) to end
      - Click back to Proposals. Confirm that the proposal is visible in Inactive section within 30 seconds.
  - Regression: Check /dydx to confirm non-cosmos communities are unaffected

## Deployment Plan

1. 

## Other Considerations

- In testing, I noticed the proposal voting period countdown was wrong. https://github.com/hicommonwealth/commonwealth/issues/4974
